### ISSUE TITLE
JSON.stringify incorrectly serializes ExpandoObject when it contains a `Values` field

### ISSUE THREAD

**Issue by sfmskywalker** [sebastienros/jint#995]
## TLDR;

- ExpandoObjects containing a `Values` key don't get stringified properly.

**Expected result**
```json
{"Values":["Foo","Bar","Baz"]}
```

**Actual result**
```json
{"Values":{"Count":1,"IsReadOnly":true}}
```

## Steps to reproduce:
Take the following JavaScript script:

```javascript
({ Values: ["Foo", "Bar", "Baz"] })
```

When evaluated, it will return an `ExpandoObject` containing a `Values` key and an array value.
So far, so good.

> We could also just use the following code to create an expando object of course:
> ```csharp
> dynamic dynamicObject = new ExpandoObject();
> dynamicObject.Values = new[] { "Foo", "Bar", "Baz" };
> ```

When we now `JSON.stringify` this expando object, the result will be:

```json
{"Values":{"Count":1,"IsReadOnly":true}}
```

Instead of the expected:

```json
{"Values":["Foo","Bar","Baz"]}
```

Interestingly enough, when I change the `Values` key to e.g. `MyValues`, the stringify operation does return the expected JSON:

```json
{"MyValues":["Foo","Bar","Baz"]}
```

I attached a small console app demonstrating the issue. It also shows that using `Values` as a key does not cause any issues when using Newtonsoft.Json's `JsonConvert` to serialize the expando object:

![image](https://user-images.githubusercontent.com/938393/139838989-08a38f4b-9b85-4aec-9c81-4fbecd9815e0.png)

Sample project:
[JintExpandoTest.zip](https://github.com/sebastienros/jint/files/7460251/JintExpandoTest.zip)

**Comment by lahma** [comment]
Did you try with latest MyGet version?

**Comment by sfmskywalker** [comment]
I did not, but will do 👍🏻

**Comment by sfmskywalker** [comment]
I just updated to the latest Jint package (`1404099859.0.0`) but no luck unfortunately. The result remains `{"Values":{"Count":1,"IsReadOnly":true}}`.

**Comment by lahma** [comment]
Thanks for confirming!

**Comment by sfmskywalker** [comment]
Sweet! I'll try it out once the build completes.

**Comment by sfmskywalker** [comment]
Works like a charm. Thanks for the quick fix 💪🏻
